ci-operator/step-registry/smongiar/deploy-resources/smongiar-deploy-resources-commands.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function create_foo_project_and_policies()
+{
+  oc new-project smongiar-interop
+  oc label --overwrite ns ${1} pod-security.kubernetes.io/enforce=privileged
+  oc label --overwrite ns ${1} pod-security.kubernetes.io/enforce-version-
+  oc adm policy add-scc-to-user privileged -z default -n ${1}
+}
+
+#function create_foo_secrets()
+#{
+#  oc create -f ./quay.io-secret.yml
+#  rm -f quaycfg.json
+#  oc secrets link default quay.io --for=pull
+#  oc secrets link builder quay.io --for=pull
+#  oc secrets link deployer quay.io --for=pull
+#}
+
+function create_foo_configmaps()
+{
+  cat << EOF > settings.xml
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+	<localRepository>/deployments/.m2/repository</localRepository>
+</settings>
+EOF
+
+REGISTRY_REDHAT_IO_AUTH=$(cat /tmp/secrets/smongiar-qe/registry-redhat-io-auth-secret)
+QUAY_IO_AUTH=$(cat /tmp/secrets/smongiar-qe/quay-io-auth-secret)
+
+export REGISTRY_REDHAT_IO_AUTH
+export QUAY_IO_AUTH
+
+ cat << EOF > test.properties
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-properties
+  namespace: smongiar
+data:
+  test.properties: >
+    xtf.config.delete.namespace=true
+    xtf.config.oreg.registry=registry.redhat.io
+    xtf.config.oreg.auth=$REGISTRY_REDHAT_IO_AUTH
+    xtf.config.quay.registry=quay.io
+    xtf.config.quay.auth=$QUAY_IO_AUTH
+    xtf.openshift.admin.username=admin
+    xtf.openshift.admin.password=admin
+    xtf.openshift.master.username=admin
+    xtf.openshift.master.password=admin
+    xtf.local.prebuild=true
+    xtf.config.gitlab=true
+binaryData: {}
+EOF
+
+  oc create configmap mvn-settings -n ${1} --from-file=settings.xml
+  oc create configmap test-properties -n ${1} --from-file=test.properties
+}
+
+function create_foo_volumes()
+{
+  oc create -f - <<EOF
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: persistent-xpaas-qe
+spec:
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+  volumeMode: Filesystem
+status:
+  accessModes:
+    - ReadWriteMany
+    - ReadWriteOnce
+  capacity:
+    storage: 1Gi
+EOF
+}
+
+echo "Create project"
+create_foo_project_and_policies smongiar-interop
+
+echo "Create persistent volumes"
+create_foo_volumes
+
+#echo "Create secrets"
+#create_foo_secrets
+
+echo "Create configmaps"
+create_foo_configmaps smongiar-interop
+
+
+echo "Create xpaas-qe image stream"
+oc import-image rh_integration/xpaas-qe --from=registry.ci.openshift.org/ci/xpaas-qe:7.11.x-openjdk1.8_local -n smongiar-interop --confirm
```

Comment:
I am not sure if the container will have permission to access image from `registry.ci.openshift.org`.
May have to look into this!

Reply:
In case I should change my "*-ref.yaml" files... but I don't know how

Reply:
If I understand this correctly I might be able to shed some light on declaring the github repository. 

It allows you to build images from dockerfiles that exist in the repository during the job execution. This images will be pushed to the internal registry of the build cluster for the life of the job. Then these images can be used as the env for a ref.

Since we have already placed an image in the internal registry this dynamic functionality won't be used, I believe. Declaring the the GitHub repo is also a way to allow for pre-submit and post-submit jobs to be run when PRs are submit & merged to the repo if you were to ever desire that behavior. Without declaring a repo the automated checks for the ci-operator will fail, and working around those is a bit of an ugly process

Reply:
about the direct use of an image in the registry.ci.openshift.org (internal registry) to create a pod in the ephemeral cluster under test. This is not something I've tried yet.

I've seen comments to run pj-rehearse but I cannot find any results.

Reply:
Thanks for clarifying this, @ascerra , anyway, about the creation of a new image, it makes for us the process a little bit complicated. We maintain a proper image as you already mentioned and all the steps performed in the container are declared in our Dockerfile: https://gitlab.cee.redhat.com/jboss-fuse-qe/dockerfiles/-/blob/master/xpaas-qe/Dockerfile

About the updates, when we submit a PR to our test suite, the image is updated as well, through a Git Hook as changes are merged

Reply:
@smongiar we can run pjrehearse to see if this command `oc import-image` gets executed successfully or not. Otherwise in my opinion, we can directly use the quay image instead to create imagestream.